theano/configdefaults.py

```diff
@@ -860,3 +868,570 @@ def good_seed_param(seed):
+    else:
+        try:
+            os.makedirs(path, 0o770)  # read-write-execute for user and group
+        except OSError as e:
+            # Maybe another parallel execution of theano was trying to create
+            # the same directory at the same time.
+            if e.errno != errno.EEXIST:
+                raise ValueError(
+                    "Unable to create the compiledir directory"
+                    " '%s'. Check the permissions." % path)
+
+    # PROBLEM: sometimes the initial approach based on
+    # os.system('touch') returned -1 for an unknown reason; the
+    # alternate approach here worked in all cases... it was weird.
+    # No error should happen as we checked the permissions.
+    init_file = os.path.join(path, '__init__.py')
+    if not os.path.exists(init_file):
+        try:
+            open(init_file, 'w').close()
+        except IOError as e:
+            if os.path.exists(init_file):
+                pass  # has already been created
+            else:
+                e.args += ('%s exist? %s' % (path, os.path.exists(path)),)
+                raise
+    return path
+
+
+def get_home_dir():
+    """
+    Return location of the user's home directory.
+
+    """
+    home = os.getenv('HOME')
+    if home is None:
+        # This expanduser usually works on Windows (see discussion on
+        # theano-users, July 13 2010).
+        home = os.path.expanduser('~')
+        if home == '~':
+            # This might happen when expanduser fails. Although the cause of
+            # failure is a mystery, it has been seen on some Windows system.
+            home = os.getenv('USERPROFILE')
+    assert home is not None
+    return home
+
+
+# On Windows we should avoid writing temporary files to a directory that is
+# part of the roaming part of the user profile. Instead we use the local part
+# of the user profile, when available.
+if sys.platform == 'win32' and os.getenv('LOCALAPPDATA') is not None:
+    default_base_compiledir = os.path.join(os.getenv('LOCALAPPDATA'), 'Theano')
+else:
+    default_base_compiledir = os.path.join(get_home_dir(), '.theano')
+
+
+AddConfigVar(
+    'base_compiledir',
+    "platform-independent root directory for compiled modules",
+    ConfigParam(
+        default_base_compiledir,
+        filter=filter_base_compiledir,
+        allow_override=False),
+    in_c_key=False)
+
+def default_compiledir():
+    return os.path.join(
+        theano.config.base_compiledir,
+        default_compiledirname())
+
+AddConfigVar(
+    'compiledir',
+    "platform-dependent cache directory for compiled modules",
+    
+    ConfigParam(
+        default_compiledir,
+        filter=filter_compiledir,
+        allow_override=False),
+    in_c_key=False)
+
+# Check if there are remaining flags provided by the user through THEANO_FLAGS.
+if THEANO_FLAGS_DICT:
+    raise Exception('theano does not recognise this flag: {}'
+                    .format(THEANO_FLAGS_DICT.keys()[0]))
```

Comment:
This should not raise an execption, but print a warning.  It should also do so for every remaining key.